### PR TITLE
feat: ✨ Implemented sendTransaction method.

### PR DESCRIPTION
**Pull request by kaumudpa**
We need some assistance here - 

We have successfully implemented `sendTransaction` which is one of the most important methods for EVM's but we are facing one minor issue.

Even if we send the transaction as `legacy` it still is sent and processed as `eip1559`.

@redDwarf03 Can you please look into this.
@Chralu May be some implementation is wrong at our end.

Can you please assist?

### CODE REVIEW

**Comment by Chralu** [comment]
Hi @kaumudpa 

I take a look at it.
In the meantime, you could get inspiration from https://github.com/archethic-foundation/wagmi_flutter_web/pull/13.

**Comment by kaumudpa** [comment]
We did looked into #13 Our issue with `sendTransaction` is totally different!

**Comment by Chralu** [comment]
@kaumudpa 

When I run your example, it seems to work : Js object is a 'legacy' type as expected and Tx is OK.

What makes you think the type is wrong ?

**Comment by kaumudpa** [comment]
Yes - It looks like you checked this on Amoy - Try checking the same on Sepolia and notice the difference.

Some issue with Amoy i guess that even if we try to do a eip1559 transaction on amoy it mines it as a legacy transaction only.

Rest everywhere even if we try to do this as legacy it mines it as eip1559 only.

**Comment by kaumudpa** [comment]
@Chralu

Even we are confused with this behaviour, But please do give it a try. We tested it on Sepolia and even Polygon Mainnet.

**Comment by Chralu** [comment]
@redDwarf03 will be more efficient to check/resolve that Amoy issue.

About the code itself, could you update your parameters implementation accordingly to [that commit](https://github.com/archethic-foundation/wagmi_flutter_web/pull/13/commits/55d8434ce7b942ca096b7f64b591d8d60829fddc) ? It would be great 🙏

**Comment by kaumudpa** [comment]
Will see abt the code and push it asap.
In the mean time try sending an eip1559 transaction on amoy - it will be mined as legacy.

And on all other chains including eth, sepolia and polygon mainchain - Any transaction sent be it legacy or eip1995 are always mined as eip1559 only . - No idea why!

**Comment by kaumudpa** [comment]
@Chralu & @redDwarf03 Also one more suggestion - Instead of resolving this on amoy lets try to resolve this on sepolia/Polygon Mainchain as i went through 100+ random transactions on Amoy - all of them are getting mined as legacy (Alteast on Amoy block scanner site) - I have not seen a single eip1559 transaction there yet.

**Comment by kaumudpa** [comment]
@Chralu code is now arrange as per your request.

**Comment by redDwarf03** [comment]
When a transaction is sent on Polygon using the "legacy" transaction format, it is sometimes processed under the EIP-1559 format. This is due to how the Polygon network and compatible nodes handle transactions.
The Polygon network is compatible with older transaction formats (legacy) while also supporting EIP-1559. If a transaction is sent in legacy format, the nodes can still process it, but the processing mechanism under the hood might adapt it to the new format to take advantage of the benefits of EIP-1559, such as better fee management.
Since the introduction of EIP-1559, which redefined the fee structure and introduced concepts like the "base fee" and the "tip," many networks have adopted this standard to improve network efficiency and user experience. Even if you send a transaction in legacy format, miners and validators may prefer to process it under the EIP-1559 structure to take advantage of these improvements.
And the network nodes may automatically convert legacy transactions into EIP-1559 transactions to simplify fee management and improve the efficiency of transaction processing on the network.

In summary, even though you send a transaction in legacy format, the Polygon network may choose to mine it under EIP-1559 due to compatibility and the advantages offered by this new format. This generally does not affect the outcome of your transaction but can improve fee efficiency and predictability.

Your PR works fine i think. 
Could you rebase your PR from main and i'll merge it.
Thx

**Comment by kaumudpa** [comment]
> When a transaction is sent on Polygon using the "legacy" transaction format, it is sometimes processed under the EIP-1559 format. This is due to how the Polygon network and compatible nodes handle transactions. The Polygon network is compatible with older transaction formats (legacy) while also supporting EIP-1559. If a transaction is sent in legacy format, the nodes can still process it, but the processing mechanism under the hood might adapt it to the new format to take advantage of the benefits of EIP-1559, such as better fee management. Since the introduction of EIP-1559, which redefined the fee structure and introduced concepts like the "base fee" and the "tip," many networks have adopted this standard to improve network efficiency and user experience. Even if you send a transaction in legacy format, miners and validators may prefer to process it under the EIP-1559 structure to take advantage of these improvements. And the network nodes may automatically convert legacy transactions into EIP-1559 transactions to simplify fee management and improve the efficiency of transaction processing on the network.
> 
> In summary, even though you send a transaction in legacy format, the Polygon network may choose to mine it under EIP-1559 due to compatibility and the advantages offered by this new format. This generally does not affect the outcome of your transaction but can improve fee efficiency and predictability.
> 
> Your PR works fine i think. Could you rebase your PR from main and i'll merge it. Thx

Any source for this?

Cause If you try to do the same thing with ethers.js on nodejs - you can see your transactions getting mined in the format you pass in the argument - A legacy transaction with etherejs is always mined as legacy and a eip1559 is always mined as eip1559.

**Comment by kaumudpa** [comment]
All right guys, I went down the rabbit hole and here are some important findings regarding this.

**1. Polygon's Amoy Scanner website has some Frontend issues.** - All the transactions polygonscan of amoy are shown as legacy but infact they are not. So for example consider this transaction https://amoy.polygonscan.com/tx/0x132f2be9e3e9073e4736f439cfd4c81cee4a0d7cc6616e35fe493f5ce21ae227 using alternate blockscanner site gives us the right result https://www.oklink.com/amoy/tx/0x132f2be9e3e9073e4736f439cfd4c81cee4a0d7cc6616e35fe493f5ce21ae227

**2. Transaction Type and Gas Price is defined by Connector (Connected Wallet) in our case** - Since our plugin connects to a wallet the transaction submission, trx type and gas prices are responsibilities of the connected wallet. So no matter what you pass as the trx type, gasPrice, PriorityGasPrice etc the connected wallet will always overwrite it as it deems fit before we appove the transaction. So @redDwarf03 you are right in some way when you say this PR is ready to be merged. But do remeber that you are wrong in regards to the under the hood workings of an EVM - If we would have used an RPC endpoint instead of Connected Wallet The transactions would have behaved exactly the way we would have defined them, but in our case a wallet (MetaMask) sits in between our transactions and the EVM so wallet (MetaMask) actually decides Gas Prices and trxType & not our plugin.

**Comment by kaumudpa** [comment]
Transferred to #22
